src/index.ts

```diff
@@ -88,5 +89,14 @@ export {
 export {PassThroughClient} from './auth/passthrough';
 export {DefaultTransporter} from './transporters';
 
+/**
+ * A union type for all AuthClients.
+ *
+ * @experimental
+ */
+export type AuthClients = InstanceType<
```

Comment:
Not too sure about the name; `AuthClients` could imply `AuthClient[]`. Would something like `AnyAuthClient` be clearer?

Reply:
I see what you mean about the naming. `AnyAuthClient` works for me, or `UnionAuthClients`? this is a tough naming situation.

My other feedback is that there is a lot going on in the `InstanceType` and the `Extract` - is it possible to split this up into a couple of lines and to comment what's going on so it's easier to read? It's hard to parse just by looking at it

Reply:
Thanks, updated